storage/remote/queue_manager.go

```diff
@@ -266,6 +272,48 @@ func (t *QueueManager) Stop() {
 	level.Info(t.logger).Log("msg", "Remote storage stopped.")
 }
 
+// Keep track of which series we know about for lookups when sending samples to remote.
+func (t *QueueManager) storeSeries(series []tsdb.RefSeries) {
+	for i := 0; i < len(series); i++ {
+		var ls = make(model.LabelSet)
+		for _, label := range series[i].Labels {
+			ls[model.LabelName(label.Name)] = model.LabelValue(label.Value)
+		}
+		_, ok := t.series[series[i].Ref]
+		if !ok {
+			t.processExternalLabels(ls, series[i].Labels)
+			relabel.Process(ls, t.relabelConfigs...)
+			t.series[series[i].Ref] = labelsToLabelsProto(labelsetToLabels(ls))
```

Comment:
Consider combining `labelsToLabelsProto` and `labelsetToLabels` into a single function.

Also, labels are sorted, which is lost when you convert to model.LabelSet.  Need to sort again.

Reply:
Do you mean they have to be sorted in order to be treated as the same series?

Reply:
Yes, thats what I mean.

Reply:
:+1: makes sense. I am sorting the labels in order of `LabelName` in `labelsetToLabels`. I can look at combining `labelsToLabelsProto` and `labelsetToLabels`, but `labelsToLabelsProto` is also used by `ToQueryResult` and the change seems low priority for now.